test/select.jl

```diff
@@ -1306,4 +1327,246 @@ end
+                          :a => (x -> (k=x, l=x.+1)) => AsTable,
+                          :a => ByRow(x -> (s=x, t=x+1)) => :z) ==
+                  DataFrame(a_function=[[1, 2], [2, 3]], x1=[1, 2], x2=[2, 3],
+                            p=[1, 2], q=[2, 3], s=[1, 2], t=[2, 3], k=[1, 2], l=[2, 3],
+                            z=[(s=1, t=2), (s=2, t=3)])
+            @test select(df, :a => ByRow(x -> [x,x+1]),
+                         :a => ByRow(x -> [x, x+1]) => AsTable,
+                         :a => ByRow(x -> [x, x+1]) => [:p, :q],
+                         :a => ByRow(x -> (s=x, t=x+1)) => AsTable,
+                         :a => (x -> (k=x, l=x.+1)) => AsTable,
+                         :a => ByRow(x -> (s=x, t=x+1)) => :z) ==
+                  DataFrame(a_function=[[1, 2], [2, 3]], x1=[1, 2], x2=[2, 3],
+                            p=[1, 2], q=[2, 3], s=[1, 2], t=[2, 3], k=[1, 2], l=[2, 3],
+                            z=[(s=1, t=2), (s=2, t=3)])
+            @test transform(df, :a => ByRow(x -> [x,x+1]),
+                            :a => ByRow(x -> [x, x+1]) => AsTable,
+                            :a => ByRow(x -> [x, x+1]) => [:p, :q],
+                            :a => ByRow(x -> (s=x, t=x+1)) => AsTable,
+                            :a => (x -> (k=x, l=x.+1)) => AsTable,
+                            :a => ByRow(x -> (s=x, t=x+1)) => :z) ==
+                  [df DataFrame(a_function=[[1, 2], [2, 3]], x1=[1, 2], x2=[2, 3],
+                                p=[1, 2], q=[2, 3], s=[1, 2], t=[2, 3], k=[1, 2], l=[2, 3],
+                                z=[(s=1, t=2), (s=2, t=3)])]
+            @test_throws ArgumentError select(df, :a => (x -> [(a=1,b=2), (a=1, b=2, c=3)]) => AsTable)
+            @test_throws ArgumentError select(df, :a => (x -> [(a=1,b=2), (a=1, c=3)]) => AsTable)
+            @test_throws ArgumentError combine(df, :a => (x -> (a=1,b=2)) => :x)
+        end
+    end
+
+    @testset "check correctness of duplicate column names" begin
+        for df in (DataFrame(a=1:2, b=3:4, c=5:6), view(DataFrame(a=1:3, b=3:5, c=5:7, d=11:13), 1:2, 1:3))
+            @test select(df, :b, :) == DataFrame(b=3:4, a=1:2, c=5:6)
+            @test select(df, :b => :c, :) == DataFrame(c=3:4, a=1:2, b=3:4)
+            @test_throws ArgumentError select(df, :b => [:c, :d], :)
+            @test_throws ArgumentError select(df, :a, :a => x -> (a=[1,2], b=[3,4]))
+            @test_throws ArgumentError select(df, :a, :a => (x -> (a=[1,2], b=[3,4])) => AsTable)
+            @test select(df, [:b, :a], :a => (x -> (a=[11,12], b=[13,14])) => AsTable, :) ==
+                  DataFrame(b=[13, 14], a=[11, 12], c=[5, 6])
+            @test select(df, [:b, :a], :a => (x -> (a=[11,12], b=[13,14])) => [:b, :a], :) ==
+                  DataFrame(b=[11, 12], a=[13, 14], c=[5, 6])
+        end
+    end
+end
+
+@testset "empty ByRow" begin
+    df = DataFrame(a=1:3)
+
+    @test select(df, [] => ByRow(() -> 1)) == DataFrame("function" => [1, 1, 1])
```

Comment:
Ah, so the default name we choose may not be the best idea. Should we use e.g. "fun" instead?

Reply:
not sure - I think this case is very rare in practice.

@pdeffebach - any opinion?

Reply:
@pdeffebach - in case you missed it. Would you prefer `function` or `fun` in DataFramesMeta.jl (and in general) as a default in this case? For me `function` is more explicit (that is why I have chosen it), but it has a shortcoming that `function` is a reserved word (which is mildly inconvenient, as you can write `df.function` the only thing you cannot write is `DataFrame(function  = 1)` as it will error)

Reply:
Yeah it shouldn't be a very common use.